resdk/resources/data.py

```diff
@@ -28,27 +28,34 @@ class Data(BaseResolweResource):
 
     endpoint = 'data'
 
-    #: (lazy loaded) annotated ``Sample`` to which ``Data`` object belongs
-    _sample = None
-    #: (lazy loaded) list of collections to which data object belongs
-    _collections = None
-
-    WRITABLE_FIELDS = ('descriptor_schema', 'descriptor',
-                       'tags') + BaseResolweResource.WRITABLE_FIELDS
-    UPDATE_PROTECTED_FIELDS = ('input', 'process') + BaseResolweResource.UPDATE_PROTECTED_FIELDS
-    READ_ONLY_FIELDS = ('process_input_schema', 'process_output_schema', 'output', 'started',
-                        'finished', 'checksum', 'status', 'process_progress', 'process_rc',
-                        'process_info', 'process_warning', 'process_error', 'process_type',
-                        'process_name') + BaseResolweResource.READ_ONLY_FIELDS
+    READ_ONLY_FIELDS = BaseResolweResource.READ_ONLY_FIELDS + (
```

Comment:
Add a test that checks that fields in READ_ONLY_FIELDS, UPDATE_PROTECTED_FIELDS and WRITABLE_FIELDS are defined in the resource classes.

Reply:
Done.